harness/src/interface.rs

```diff
@@ -48,7 +48,11 @@ impl Interface {
     /// Read messages out of the raft.
     pub fn read_messages(&mut self) -> Vec<Message> {
         match self.raft {
-            Some(_) => self.msgs.drain(..).collect(),
+            Some(_) => {
+                let mut msgs = self.msgs.drain(..).collect::<Vec<_>>();
```

Comment:
You might be able to avoid reallocation here by combining `drain` and `chain`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=47f3604fa7d8fa5cbe9b3105d47b4d68